### PR TITLE
[PkgConfigDeps] Added new property `component_version`

### LINKED ISSUE #10629: [bug] PkgConfigDeps: component version not honored
### Environment Details (include every applicable attribute)
  * Operating System+version: macOS Monterey
  * Compiler+version: apple-clang 13.0
  * Conan version: 1.45.0
  * Python version: 3.9.10

### Steps to reproduce (Include if Applicable)

- `conan install ffmpeg/4.4@ -g PkgConfigDeps`
- inspect pc files, all versions are set to 4.4 => not good
- `conan install ffmpeg/4.4@ -g pkg_config`
- inspect pc files, many versions are different because ffmpeg recipe set version of each component with a custom value => good

### Logs (Executed commands with output) (Include/Attach if Applicable)

For example here the content of `libavdevice.pc` (avdevice is a component of ffmpeg)

from `PkgConfigDeps` generator:

```
prefix=/Users/spaceim/.conan/data/ffmpeg/4.4/_/_/package/90fa4cf4928013c69bdbdd0bf4bac261ff5e5328
libdir1=${prefix}/lib
includedir1=${prefix}/include

Name: libavdevice
Description: Conan component: libavdevice
Version: 4.4
Libs: -L"${libdir1}" -lavdevice -framework CoreFoundation -framework Foundation -framework CoreGraphics -framework OpenGL -framework AppKit -framework AVFoundation -framework CoreAudio -F /Users/charpentiera/.conan/data/ffmpeg/4.4/_/_/package/90fa4cf4928013c69bdbdd0bf4bac261ff5e5328/Frameworks 
Cflags: -I"${includedir1}" 
Requires: libavutil libavfilter libswscale libavformat libavcodec libswresample libpostproc
```

from `pkg_config` generator:

```
prefix=/Users/spaceim/.conan/data/ffmpeg/4.4/_/_/package/90fa4cf4928013c69bdbdd0bf4bac261ff5e5328
libdir=${prefix}/lib
includedir=${prefix}/include

Name: ffmpeg-libavdevice
Description: Conan package: ffmpeg-libavdevice
Version: 58.13.100
Libs: -L"${libdir}" -lavdevice  -Wl,-rpath,"${libdir}" -framework CoreFoundation -framework Foundation -framework CoreGraphics -framework OpenGL -framework AppKit -framework AVFoundation -framework CoreAudio
Cflags: -I"${includedir}"
Requires: libavutil libavfilter libswscale libavformat libavcodec libswresample libpostproc
```

### PR DESCRIPTION
**Pull request by franramirez688**
Changelog: Feature: Added new property `component_version` for `PkgConfigDeps` and legacy `PkgConfig`.
Changelog: Feature: Changed `.pc` file description field for components in `PkgConfigDeps`.
Closes: https://github.com/conan-io/conan/issues/10629
Docs: https://github.com/conan-io/docs/pull/2433

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

File: conan/tools/gnu/pkgconfigdeps/pc_files_templates.py [hunk 1]
```diff
@@ -88,8 +88,8 @@ def _get_formatted_dirs(folders, prefix_path_):
 
 def get_pc_filename_and_content(conanfile, dep, name, requires, description, cpp_info=None):
     package_folder = dep.package_folder
-    version = dep.ref.version
     cpp_info = cpp_info or dep.cpp_info
+    version = cpp_info.get_property("pkg_config_version") or dep.ref.version
```
Comment (memsharded) [inline conan/tools/gnu/pkgconfigdeps/pc_files_templates.py]: better ``component_version`` just in case some other build system wants it too.

File: conan/tools/gnu/pkgconfigdeps/pc_files_creator.py [hunk 2]
```diff
@@ -33,15 +33,15 @@ def _get_aliases_pc_files_and_content(conanfile, dep, aliases_info, root_pc_file
...
+def _get_components_pc_files_and_content(conanfile, dep, package_info, components_info):
     """
     Get the PC files and content for dependency's components
     """
     pc_files = {}
     # Loop through all the package's components
     for pc_info_component in components_info:
         # Get the *.pc file content for each component
-        description = "Conan component: %s" % pc_info_component.name
+        description = "Conan component: %s-%s" % (package_info.name, pc_info_component.name)
```
Comment (memsharded) [inline conan/tools/gnu/pkgconfigdeps/pc_files_creator.py:44]: Was this a bug? why this change?

Comment (franramirez688) [reply conan/tools/gnu/pkgconfigdeps/pc_files_creator.py:44]: No, it wasn't. Old PkgConfig was showing `[pkg_name]-[component_name]` and I realized that `PkgConfigDeps` is not doing that right now. It's only to keep the same logic.

**Comment by SpaceIm** [comment]
Currently, `pkg_config` generator seems to listen `pkg_config_custom_content` as well: https://github.com/conan-io/conan/blob/a43c1eb782b96fc5d65a054dced719a037974a52/conans/client/generators/pkg_config.py#L87-L89

Would it make sense to also backport this property into `pkg_config` generator?
I guess it would be as simple as:

```patch
-        lines.append("Version: %s" % cpp_info.version)
+        version = cpp_info.get_property("pkg_config_version") or cpp_info.version
+        lines.append("Version: {}".format(version))
```

**Review by memsharded — APPROVED** [review]
